Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceRenderer.swift

```diff
@@ -12,41 +12,136 @@ import UIKit
+    func dismissExperience(experienceID: String?, markComplete: Bool, completion: ((Result<Void, Error>) -> Void)?)
 }
 
 @available(iOS 13.0, *)
 internal class ExperienceRenderer: ExperienceRendering {
 
-    private let stateMachine: ExperienceStateMachine
-    private let analyticsObserver: ExperienceStateMachine.AnalyticsObserver
+    private weak var container: DIContainer?
+
+    // this is the shared machine, used for modal experiences - only supports one experience
+    // running at a time
+    private let sharedStateMachine: ExperienceStateMachine
+
+    // this the mapping of experienceID -> state machine, used for non-modal experiences
+    // that can have more than one running in parallel
+    private var experienceStateMachines: [String: ExperienceStateMachine] = [:]
+
     private weak var appcues: Appcues?
     private let config: Appcues.Config
 
     init(container: DIContainer) {
+        self.container = container
         self.appcues = container.owner
         self.config = container.resolve(Appcues.Config.self)
 
         // two items below are not registered/resolved directly from container as they
         // are considered private implementation details of the ExperienceRenderer - helpers.
-        self.stateMachine = ExperienceStateMachine(container: container)
-        self.analyticsObserver = ExperienceStateMachine.AnalyticsObserver(container: container)
+        self.sharedStateMachine = ExperienceStateMachine(container: container)
     }
 
     func show(experience: Experience, priority: RenderPriority, published: Bool, completion: ((Result<Void, Error>) -> Void)?) {
+        guard let container = container else { return }
+
+        let stateMachine: ExperienceStateMachine
+
+        if experience.isNonModal {
+            // if the experience is non-modal, get or create the state machine for rendering
+            if let experienceStateMachine = experienceStateMachines[experience.instanceID.uuidString] {
+                stateMachine = experienceStateMachine
+            } else {
+                stateMachine = ExperienceStateMachine(container: container)
+                stateMachine.addObserver(
+                    StateMachineCleanupObserver(experienceRenderer: self, experienceID: experience.instanceID.uuidString))
+                experienceStateMachines[experience.instanceID.uuidString] = stateMachine
+            }
+        } else {
+            // if it is modal, use the shared state machine
+            stateMachine = sharedStateMachine
+        }
+
+        show(experience: experience, in: stateMachine, priority: priority, published: published, completion: completion)
+    }
+
+    func show(qualifiedExperiences: [Experience], priority: RenderPriority, completion: ((Result<Void, Error>) -> Void)?) {
+        let nonModal = qualifiedExperiences.filter { $0.isNonModal }
+        let modal = qualifiedExperiences.filter { !$0.isNonModal }
```

Comment:
I know performance isn't a big concern for a list of experiences since it's small, but you could do this:

```swift
extension Collection {
    func separate(predicate: (Iterator.Element) -> Bool) -> (matching: [Iterator.Element], notMatching: [Iterator.Element]) {
        var separated: ([Iterator.Element],[Iterator.Element]) = ([],[])
        for element in self {
            if predicate(element) {
                separated.0.append(element)
            } else {
                separated.1.append(element)
            }
        }
        return separated
    }
}

let (nonModal, modal) = qualifiedExperiences.separate { $0.isNonModal }
```